std-bits/table/src/main/java/org/enso/table/parsing/IncrementalDatatypeParser.java

```diff
@@ -0,0 +1,58 @@
+package org.enso.table.parsing;
+
+import org.enso.table.data.column.builder.object.Builder;
+import org.enso.table.data.column.storage.Storage;
+import org.enso.table.data.column.storage.StringStorage;
+import org.enso.table.parsing.problems.ProblemAggregator;
+import org.enso.table.read.WithProblems;
+
+/**
+ * A base type for a datatype parsing strategy which relies on a method parsing a single value.
+ *
+ * <p>It specifies the strategy for parsing text cells into some target type, reporting issues and
+ * building the resulting table column.
+ */
+public abstract class IncrementalDatatypeParser implements DatatypeParser {
+
+  /**
+   * Parses a single cell.
+   *
+   * @param text the text contents to parse, it will never be null in the default implementation -
+   *     null values are just passed as-is without any parsing attempts by default
+   * @param problemAggregator an instance of the problem aggregator, used for reporting parsing
+   *     problems
+   * @return the parsed value or null if the value could not be parsed or could be parsed but should
+   *     be treated as missing value
+   */
+  public abstract Object parseSingleValue(String text, ProblemAggregator problemAggregator);
```

Comment:
This feels and internal method

The public one should probably exist in due course.